README.md

```diff
@@ -1,41 +1,158 @@
-   will be mounted inside the container under `~/home`.
-5. Once inside the container, run `install-dev.sh` under `analyzer/` to build
-   and install the Habitat package.
-6. In your scripts, `import habitat` to get access to Habitat. See
-   `experiments/run_experiment.py` for an example showing how to use Habitat.
+Habitat is a tool that predicts a deep neural network's training iteration execution time on a given GPU. It currently supports PyTorch. To learn more about how Habitat works, please see our [research paper](https://arxiv.org/abs/2102.00527).
 
-**Note:** Habitat needs access to your GPU's performance counters, which
-requires special permissions if you are running with a recent driver (418.43 or
-later). If you encounter a `CUPTI_ERROR_INSUFFICIENT_PRIVILEGES` error when
-running Habitat, please follow the instructions
-[here](https://developer.nvidia.com/ERR_NVGPUCTRPERM)
-and in [issue #5](https://github.com/geoffxy/habitat/issues/5).
+<h2 id="installation">Installation</h2>
 
+To run Habitat, you need:
+- [Python 3.6+](https://www.python.org/)
+- [Pytorch 1.1.0+](https://pytorch.org/)
+- A system equiped with an Nvidia GPU.
 
-## License
+Currently, we have predictors for the following Nvidia GPUs:
+
+| GPU        | Generation  | Memory | Mem. Type | SMs |
+| ---------- |:-----------:| ------:| :-------: | :-: |
+| P4000      | Pascal      | 8 GB   | GDDR5     | 14  |
+| P100       | Pascal      | 16 GB  | HBM2      | 56  |
+| V100       | Volta       | 16 GB  | HBM2      | 80  |
+| 2070       | Turing      | 8 GB   | GDDR6     | 36  |
+| 2080Ti     | Turing      | 11 GB  | GDDR6     | 68  |
+| T4         | Turing      | 16 GB  | GDDR6     | 40  |
+| 3090       | Ampere      | 24 GB  | GDDR6X    | 82  |
+
+**NOTE:** Not implmented yet
+```zsh
+python3 -m pip install habitat
+python3 -c "import habitat"
+```
+
+<h2 id="build">Building from source</h2>
+
+Prerequsites:
+- A system equiped with an Nvidia GPU with properly configured CUDA
+- [CUDA Toolkit](https://developer.nvidia.com/cuda-toolkit-archive)
+- [cmake v3.17+](https://github.com/Kitware/CMake/releases)
+- [Habitat pre-trained models](https://zenodo.org/record/4876277)
+
+```zsh
+git clone https://github.com/CentML/habitat.git && cd habitat
+git submodule init && git submodule update
+
+# Download the pre-trained models
+cd analyzer
+curl -O https://zenodo.org/record/4876277/files/habitat-models.tar.gz\?download\=1
+
+# Install the models
+./extract-models.sh
+```
+
+**Note:** Habitat needs access to your GPU's performance counters, which requires special permissions if you are running with a recent driver (418.43 or later). If you encounter a `CUPTI_ERROR_INSUFFICIENT_PRIVILEGES` error when running Habitat, please follow the instructions [here](https://developer.nvidia.com/ERR_NVGPUCTRPERM) and in [issue #5](https://github.com/geoffxy/habitat/issues/5).
+
+### Building with Docker
+
+1. Run `setup.sh` under `docker/` to build the Habitat container image.
```

Comment:
Again, something to think about - I took a look at the Dockerfile under `docker/`. Would it be easier to use [NGC PyTorch containers](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch) as your base image? This way you don't have to manually align the version (of CUDA and any other NVIDIA tools) in the image with the version that PyTorch needs going forward.
NGC PyTorch containers have a fair amount of enterprise users. It'd be nice if we have a list in the readme that says "Habitat is tested on NGC PyTorch containers from version X to version Y".

Reply:
That's a good idea. I'll try building Habitat using these containers.

Reply:
I've tested habitat on the NGC containers and it works.

I think we should still use the nvidia/cuda containers since we can manually specify the CUDA version. This is useful because we want to support users with all relevant CUDA versions and not just the newest ones. The NGC container environment would still be covered as one of the cases (Python3.8/CUDA 11.7).

Reply:
I've tried Habitat with the NGC containers and it works out of the box. The user should either:
* Clone the repository, initialize the submodules, run `install-dev.sh` then import the saved models, or
* Install the correct wheel file (in the case of the latest NGC container, CUDA 11.7 + Python 3.8)

With that said, we should still use `nvidia/cuda` images for automated builds since would allow us to specify the version of CUDA that's linked to habitat_cuda.